Example/Database/Tests/Integration/FData.m

```diff
@@ -2873,6 +2874,137 @@ - (void)testServerValuesTransactionLocalEvents {
                 @"Number should be no more than 2 seconds ago");
 }
 
+- (void)testServerIncrementOverwritesExistingData {
+  FIRDatabaseReference *ref = [FTestHelpers getRandomNode];
+  __block NSMutableArray *found = [NSMutableArray new];
+  NSMutableArray *expected = [NSMutableArray new];
+  [ref observeEventType:FIRDataEventTypeValue
+              withBlock:^(FIRDataSnapshot *snap) {
+                [found addObject:snap.value];
+              }];
+
+  // Going offline ensures that local events get queued up before server events
+  [ref.repo interrupt];
+
+  // null + incr
+  [ref setValue:[FIRServerValue increment:@1]];
+  [expected addObject:@1];
+
+  // number + incr
+  [ref setValue:@5];
+  [ref setValue:[FIRServerValue increment:@1]];
+  [expected addObject:@5];
+  [expected addObject:@6];
+
+  // string + incr
+  [ref setValue:@"hello"];
+  [ref setValue:[FIRServerValue increment:@1]];
+  [expected addObject:@"hello"];
+  [expected addObject:@1];
+
+  // object + incr
+  [ref setValue:@{@"hello" : @"world"}];
+  [ref setValue:[FIRServerValue increment:@1]];
+  [expected addObject:@{@"hello" : @"world"}];
+  [expected addObject:@1];
+
+  [self waitUntil:^BOOL {
+    return found.count == expected.count;
+  }];
+  XCTAssertEqualObjects(expected, found);
+  [ref.repo resume];
+}
+
+- (void)testServerIncrementPriority {
+  FIRDatabaseReference *ref = [FTestHelpers getRandomNode];
+  __block NSMutableArray *found = [NSMutableArray new];
+  NSMutableArray *expected = [NSMutableArray new];
+  [ref observeEventType:FIRDataEventTypeValue
+              withBlock:^(FIRDataSnapshot *snap) {
+                [found addObject:snap.priority];
+              }];
+
+  // Going offline ensures that local events get queued up before server events
+  // Also necessary because increment may not be live yet in the server.
+  [ref.repo interrupt];
+
+  // null + incr
+  [ref setValue:@0 andPriority:[FIRServerValue increment:@1]];
+  [expected addObject:@1];
+  [ref setValue:@0 andPriority:[FIRServerValue increment:@1.5]];
+  [expected addObject:@2.5];
+
+  [self waitUntil:^BOOL {
+    return found.count == expected.count;
+  }];
+  XCTAssertEqualObjects(expected, found);
+  [ref.repo resume];
+}
+
+- (void)testServerIncrementOverflowAndTypeCoercion {
```

Comment:
This one doesn't make sense to have online behavior; that's totally redundant with server tests (we're trying to test local coercion)